plugin/src/test/java/org/opensearch/ml/action/models/DeleteModelTransportActionTests.java

```diff
@@ -159,20 +188,71 @@ public void testDeleteModel_Success() throws IOException {
         verify(actionListener).onResponse(deleteResponse);
     }
 
+    public void testDeleteModel_BlockedBySearchPipeline() throws IOException {
+        //org.opensearch.search.pipeline.PipelineConfiguration pipelineConfiguration = new PipelineConfiguration();
+        when(searchPipelineConfiguration.getId()).thenReturn("1");
+        when(searchPipelineConfiguration.getConfigAsMap()).thenReturn(configDataMap);
+        when(getSearchPipelineResponse.pipelines()).thenReturn(List.of(searchPipelineConfiguration));
+        doAnswer(invocation -> {
+            ActionListener<GetSearchPipelineResponse> listener = invocation.getArgument(2);
+            listener.onResponse(getSearchPipelineResponse);
+            return null;
+        }).when(client).execute(eq(GetSearchPipelineAction.INSTANCE), any(), any());
+
+        deleteModelTransportAction.doExecute(null, mlModelDeleteRequest, actionListener);
+        ArgumentCaptor<Exception> argumentCaptor = ArgumentCaptor.forClass(Exception.class);
+        verify(actionListener).onFailure(argumentCaptor.capture());
+        assertEquals("1 search pipelines are still using this model, please delete or update the pipelines first: [1]", argumentCaptor.getValue().getMessage());
+    }
```

Comment:
In this unit test, can you create another search pipeline which is not using the model id. Also please update testDeleteModel_BlockedByIngestPipeline and testDeleteModel_BlockedByAgent accordingly